grafana/data_source_dashboards.go

```diff
@@ -0,0 +1,126 @@
+package grafana
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DatasourceDashboards() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+Datasource for retrieving all dashboards. Specify list of folder IDs to search in for dashboards.
+
+* [Official documentation](https://grafana.com/docs/grafana/latest/dashboards/)
+* [Folder/Dashboard Search HTTP API](https://grafana.com/docs/grafana/latest/http_api/folder_dashboard_search/)
+* [Dashboard HTTP API](https://grafana.com/docs/grafana/latest/http_api/dashboard/)
+`,
+		ReadContext: dataSourceReadDashboards,
+		Schema: map[string]*schema.Schema{
+			"folder_ids": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Numerical IDs of Grafana folders containing dashboards. Specify to filter for dashboards by folder (eg. `[0]` for General folder), or leave blank to get all dashboards in all folders.",
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+			},
+			"limit": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     5000,
+				Description: "Maximum number of dashboard search results to return.",
+			},
+			"tags": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of string Grafana dashboard tags to search for, eg. `[\"prod\"]`. Used only as search input, i.e., attribute value will remain unchanged.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"dashboards": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"title": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"uid": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"folder_title": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func hashDashboardSearchParameters(params map[string][]string) string {
+	// hash a sorted slice of all string parameters and corresponding values
+	hashOut := sha256.New()
+
+	var paramsList []string
+	for key, vals := range params {
+		paramsList = append(paramsList, key)
+		paramsList = append(paramsList, vals...)
+	}
+
+	sort.Strings(paramsList)
+	hashIn := strings.Join(paramsList, "")
+	hashOut.Write([]byte(hashIn))
+	return fmt.Sprintf("%x", hashOut.Sum(nil))[0:23]
```

Comment:
Also, you could achieve the same by doing:
```suggestion
	return fmt.Sprintf("%x", hashOut.Sum(nil))[:23]
```